### PR TITLE
SLING-11028 - Allow to assign to relative offsets

### PR DESCRIPTION
**Pull request by cschneider**

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java [hunk 1]
```diff
@@ -40,6 +40,8 @@ default Closeable createPoller(
 
     String assignTo(long offset);
     
+    String assignTo(Reset reset, long relativeOffset);
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java:92]: please add Javadoc

File: src/main/java/org/apache/sling/distribution/journal/Reset.java [hunk 2]
```diff
@@ -18,4 +18,5 @@
  */
 package org.apache.sling.distribution.journal;
 
+// Do not change the enum names. Implementation depends on the exact names.
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/distribution/journal/Reset.java:21]: Some more javadoc would be appropriate here as well.

**Review by joerghoh — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java (hunk 1 above)
Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java:92]: done

File: src/main/java/org/apache/sling/distribution/journal/Reset.java (hunk 2 above)
Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/Reset.java:21]: done

**Review by tmaret — APPROVED** [review]
LGTM

**Review by joerghoh — APPROVED** [review]

**Pull request merged**